doc/_static/skymap/showVisitSkyMap.py

```diff
@@ -0,0 +1,683 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+
+import argparse
+import matplotlib
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+import numpy as np
+from astropy import units
+from astropy.coordinates import SkyCoord
+from matplotlib.legend import Legend
+
+import lsst.afw.cameraGeom as cameraGeom
+import lsst.daf.butler as dafButler
+import lsst.geom as geom
+import lsst.log as lsstLog
+import lsst.sphgeom as sphgeom
+
+logger = lsstLog.Log.getLogger("showVisitSkyMap.py")
+
+
+def bboxToRaDec(bbox, wcs):
+    """Get the corners of a BBox and convert them to lists of RA and Dec."""
+    sphPoints = wcs.pixelToSky(geom.Box2D(bbox).getCorners())
+    ra = [float(sph.getRa().asDegrees()) for sph in sphPoints]
+    dec = [float(sph.getDec().asDegrees()) for sph in sphPoints]
+    return ra, dec
+
+
+def percent(values, p=0.5):
+    """Return a value a faction of the way between the min and max values in a
+    list.
+    """
+    m = min(values)
+    interval = max(values) - m
+    return m + p*interval
+
+
+def get_cmap(n, name="hsv"):
+    """Returns a function that maps each index in 0, 1, ..., n-1 to a distinct
+    RGB color; the keyword argument name must be a standard mpl colormap name.
+    """
+    return matplotlib.colormaps[name].resampled(n)
+
+
+def main(repo, collections, skymapName=None, tracts=None, visits=None, physicalFilters=None, bands=None,
+         ccds=None, ccdKey="detector", showPatch=False, saveFile=None, showCcds=False, visitVetoFile=None,
+         minOverlapFraction=None, trimToTracts=False):
+    if minOverlapFraction is not None and tracts is None:
+        raise RuntimeError("Must specify --tracts if --minOverlapFraction is set")
+    logger.info("Making butler for collections = {} in repo {}".format(collections, repo))
+    butler = dafButler.Butler(repo, collections=collections)
+    instrument = butler.registry.findDataset("camera").dataId["instrument"]
+    detectorSkipList = []
+    # Make a guess at the skymapName if not provided
```

Comment:
This "guess at the skymap name" part is a great idea -- good to head off that likely common error!